src/test/sushi-integration.t.sol

```diff
@@ -675,19 +737,68 @@ contract SushiIntegrationTest is TestBase {
         timelock.queueTransaction(
             address(masterchef),
             0,
-            "transferOwnership(address)",
-            abi.encode(address(user2)),
+            "transferOwnership(address,bool,bool)",
+            abi.encode(address(user2),true,false),
             block.timestamp + timelock.delay()
         );
 
         // Anyone can cage
         user1.cage(
             0,
-            "transferOwnership(address)",
-            abi.encode(address(user2)),
+            "transferOwnership(address,bool,bool)",
+            abi.encode(address(user2),true,false),
             block.timestamp + timelock.delay()
         );
         assertTrue(!join.live());
+
+        // Make sure this action can be executed and does what is expected (otherwise we have the wrong command)
+        hevm.warp(block.timestamp + timelock.delay());
+        timelock.executeTransaction(
+            address(masterchef),
+            0,
+            "transferOwnership(address,bool,bool)",
+            abi.encode(address(user2),true,false),
+            block.timestamp
+        );
+        assertEq(masterchef.owner(), address(user2));
+    }
+
+    function test_cage_queued_change3() public {
+        // Set this contract as admin of the timelock
+        hevm.store(
+            address(timelock),
+            bytes32(uint256(0)),
+            bytes32(uint256(address(this)))
+        );
+
+        // Queue up a malicious transaction - changing the rewards callback contract
+        timelock.queueTransaction(
+            address(masterchef),
+            0,
+            "",
+            abi.encodeWithSelector(MasterChefLike.set.selector, join.pid(), 100, address(user2), true),
+            block.timestamp + timelock.delay()
+        );
+
+        // Anyone can cage
+        user1.cage(
+            0,
+            "",
+            abi.encodeWithSelector(MasterChefLike.set.selector, join.pid(), 100, address(user2), true),
+            block.timestamp + timelock.delay()
+        );
+        assertTrue(!join.live());
+
+        // Make sure this action can be executed and does what is expected (otherwise we have the wrong command)
+        hevm.warp(block.timestamp + timelock.delay());
+        timelock.executeTransaction(
+            address(masterchef),
+            0,
+            "",
+            abi.encodeWithSelector(MasterChefLike.set.selector, join.pid(), 100, address(user2), true),
+            block.timestamp
+        );
+        assertEq(masterchef.rewarder(join.pid()), address(user2));
     }
 
     function testFail_cage_queued_irrelevant_change() public {
```

Comment:
Should have another test like this cover the "different pool id" irrelevant scenario.